build.sbt

```diff
@@ -284,7 +274,22 @@ lazy val unit = projectMatrix
     ),
     Test / test := (Test / test)
       .dependsOn(cli.projectRefs.map(_ / publishLocalTransitive): _*)
-      .value
+      .value,
+    Test / unmanagedSourceDirectories ++= {
+      val sourceDir = (Test / sourceDirectory).value
+      val maybeTargetScalaVersion =
+        TargetAxis
+          .targetScalaVersion(virtualAxes.value)
+          .flatMap(CrossVersion.partialVersion(_))
+      maybeTargetScalaVersion match {
+        case Some((n, m)) =>
+          Seq(
+            sourceDir / s"scala-target$n",
+            sourceDir / s"scala-target$n.$m"
```

Comment:
this allows to restrict execution of certain tests based on what the unit combination is testing against